main/typings.d.ts

```diff
@@ -10,13 +10,24 @@ export interface Activity {
   message: string;
 }
 
+export type FILTransactionProcessing = {
+  hash: string;
+  height: number;
+  timestamp: number?;
+  status: TransactionStatus?;
+  outgoing: boolean;
+  amount: string;
+  address: string;
+}
+
 export type FILTransaction = {
```

Comment:
Would something like this be a meaningful way how to avoid duplication?

```ts
export type FILTransaction = FILTransactionProcessing &
  Required<Pick<FILTransactionProcessing, 'timestamp' | 'status'>>
```

Explanation:
- The generic type `Required` converts all fields from optional (e.g. `number | undefined`) to required (`number`).

- The generic type `Pick` picks a subset of properties from the target object-like type.

- Together, they create a type with two required properties - `timestamp` and `status`. 

Personally, I would design these types the other way round - start with `FILTransaction` with all fields required, and then define `FILTransactionProcessing` by making some fields optional.

```ts
// helper
type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>

// FILTransaction with certain properties changed to optional
type FILTransactionProcessing = PartialBy<FILTransaction, 'timestamp' | 'status'>
```

BTW, should it be `FILTransactionProcessing` or `FILTransactionPending`?

Reply:
Great suggestion! I'll refactor it like this 🙏

Reply:
the first suggestion works, with the 2nd I get this error:

<img width="970" alt="Screenshot 2023-01-24 at 16 03 44" src="https://user-images.githubusercontent.com/10247/214329810-72ac30e1-f4b5-4c4d-8b6a-17b8aed8455d.png">

Implementation:

```ts
export type FILTransaction = {
  hash: string;
  height: number;
  timestamp: number;
  status: TransactionStatus;
  outgoing: boolean;
  amount: string;
  address: string;
}

// helper
type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>

// FILTransaction with certain properties changed to optional
export type FILTransactionProcessing = PartialBy<FILTransaction, 'hash' | 'timestamp' | 'status'>
```

Do you know how to get that to work? I would also prefer the 2nd way

Reply:
nvm, just needed to remove the null values 👍 

Thank you so much!